app/backend/src/tests/test_communities.py

```diff
@@ -563,6 +563,72 @@ def test_ListAdmins(testing_communities):
             )
             assert res.admin_user_ids == [user4_id, user5_id]
 
+    @staticmethod
+    def test_AddAdmin(testing_communities):
+        with session_scope() as session:
+            user4_id, token4 = get_user_id_and_token(session, "user4")
+            user5_id, _ = get_user_id_and_token(session, "user5")
+            user2_id, _ = get_user_id_and_token(session, "user2")
+            user8_id, token8 = get_user_id_and_token(session, "user8")
+            node_id = get_community_id(session, "Country 1, Region 1, City 2")
+
+        with communities_session(token8) as api:
+            with pytest.raises(grpc.RpcError) as err:
+                api.AddAdmin(communities_pb2.AddAdminReq(community_id=node_id, user_id=user2_id))
+            assert err.value.code() == grpc.StatusCode.FAILED_PRECONDITION
+            assert err.value.details() == errors.NODE_MODERATE_PERMISSION_DENIED
+
+        with communities_session(token4) as api:
+            res = api.ListAdmins(communities_pb2.ListAdminsReq(community_id=node_id))
+            assert res.admin_user_ids == [user4_id, user5_id]
+
+            with pytest.raises(grpc.RpcError) as err:
+                api.AddAdmin(communities_pb2.AddAdminReq(community_id=node_id, user_id=user8_id))
+            assert err.value.code() == grpc.StatusCode.FAILED_PRECONDITION
+            assert err.value.details() == errors.USER_NOT_MEMBER
+
+            with pytest.raises(grpc.RpcError) as err:
+                api.AddAdmin(communities_pb2.AddAdminReq(community_id=node_id, user_id=user5_id))
+            assert err.value.code() == grpc.StatusCode.FAILED_PRECONDITION
+            assert err.value.details() == errors.USER_ALREADY_ADMIN
+
+            api.AddAdmin(communities_pb2.AddAdminReq(community_id=node_id, user_id=user2_id))
+            res = api.ListAdmins(communities_pb2.ListAdminsReq(community_id=node_id))
+            assert res.admin_user_ids == [user2_id, user4_id, user5_id]
+
+    @staticmethod
+    def test_RemoveAdmin(testing_communities):
+        with session_scope() as session:
+            user4_id, token4 = get_user_id_and_token(session, "user4")
+            user5_id, _ = get_user_id_and_token(session, "user5")
+            user2_id, _ = get_user_id_and_token(session, "user2")
+            user8_id, token8 = get_user_id_and_token(session, "user8")
+            node_id = get_community_id(session, "Country 1, Region 1, City 2")
+
+        with communities_session(token8) as api:
+            with pytest.raises(grpc.RpcError) as err:
+                api.AddAdmin(communities_pb2.AddAdminReq(community_id=node_id, user_id=user2_id))
+            assert err.value.code() == grpc.StatusCode.FAILED_PRECONDITION
+            assert err.value.details() == errors.NODE_MODERATE_PERMISSION_DENIED
+
+        with communities_session(token4) as api:
+            res = api.ListAdmins(communities_pb2.ListAdminsReq(community_id=node_id))
+            assert res.admin_user_ids == [user4_id, user5_id]
```

Comment:
The test is only failing here in CI because it expects user 2 to be an admin already, but within the context of `test_RemoveAdmin` this doesn't make any sense. When this test is run in isolation, it succeeds because the preceding attempt to add user 2 AS user 8 fails as expected. Could it be failing because of incomplete cleanup from `test_AddAdmin` where the final list of admin ids does include user 2?

Reply:
Yes, I think maybe the communities don’t clean up the whole thing but reuse between tests.

Maybe you can make/use another community to make sure they are separate?

Reply:
I just added a line of cleanup to both

Reply:
So I don't remember the specifics of why the communities tests do one set up and one cleanup. Not sure if it was very slow or something. Sorry about that.